### DISCUSSION TITLE
Database performance and size limit

### DISCUSSION THREAD

**Discussion by natgross** [appwrite/appwrite#3562] [General]
How would you rate the performance of the database compared to MongoDb? I am not asking for a feature by feature comparison, but simply, say for a database with 20 collections, totaling  about 10gig of json documents, with several indices per collection; how does it compare to Mongo performance wise? (Am not factoring any clusters or sharding, just a simple single server comparison.

Also, the size limit, if any, on a database.

Thanks,
-nat

**Comment by stnguyen90** [comment]
@natgross, unfortunately, it's impossible to compare generically, I would suggest doing a POC to see how the two compare. 

Some differences to point out, though, are that Appwrite is an abstraction on top of the database so there are extra queries and whatnot Appwrite does rather than if you made a query directly to the database. However, Appwrite does have a caching layer as well so that *could* speed up queries.

Another thing to consider when using Appwrite is Appwrite saves you time from building your back end. If you're using mongodb, you would need to build a server that your front end connects to (including authentication, etc.). Appwrite shines by letting you focus on building your front end application without needing to build your own back end server.

**Reply by adepauldami** [reply]
Nice answer
